internal/bminventory/inventory.go

```diff
@@ -1653,6 +1660,23 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
 	return b.v2UpdateClusterInternal(ctx, params, NonInteractive)
 }
 
+func updateClusterNonePlatformParams(params installer.V2UpdateClusterParams, cluster *common.Cluster) {
+	if params.ClusterUpdateParams.UserManagedNetworking == nil || cluster.Platform == nil || cluster.UserManagedNetworking == nil {
+		return
+	}
+
+	if *params.ClusterUpdateParams.UserManagedNetworking && !*cluster.UserManagedNetworking {
+		if *cluster.Platform.Type == models.PlatformTypeBaremetal {
+			params.ClusterUpdateParams.Platform = &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}
+
+		} else if params.ClusterUpdateParams.Platform != nil && models.PlatformTypeBaremetal == *params.ClusterUpdateParams.Platform.Type {
+			params.ClusterUpdateParams.Platform = &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}
```

Comment:
You are basically overriding a param provided by the users because it's conflicting with another param,
I think this deserves returning bad request, @avishayt?

Reply:
I agree.  If they selected user-managed networking and baremetal that's 400.

Reply:
i updated it and now returning bed-request when `UserManagedNetworking=false and Platform=none or UserManagedNetworking=true and Platform=baremetal`